indico/modules/events/controllers/creation.py

```diff
@@ -48,14 +48,22 @@ def _process_args(self):
         self.root_category = Category.get_root()
         self.single_category = not self.root_category.children
 
+    def _has_only_subcategories(self, category):
+        return not bool(category.events) and bool(category.children)
+
     @cached_property
     def _default_category(self):
         try:
             category_id = int(request.args['category_id'])
         except (ValueError, KeyError):
-            return self.root_category if self.single_category else None
+            return None if self._has_only_subcategories(self.root_category) else self.root_category
+
+        category = Category.get(category_id, is_deleted=False)
+        if category and self._has_only_subcategories(category) and \
```

Comment:
we don't use `\` ;) better use `()` around the whole condition - either black-style or with double indentation in the wrapped line